src/coreclr/nativeaot/Runtime/thread.h

```diff
@@ -112,6 +112,10 @@ struct ThreadBuffer
 #ifdef FEATURE_GC_STRESS
     uint32_t                m_uRand;                                // current per-thread random number
 #endif // FEATURE_GC_STRESS
+
+    // The MethodTable is remembered in some slow-path allocation paths. This value is used in event tracing.
+    // It may statistically correlate with the most allocated type on the given stack/thread.
+    MethodTable*            m_LastAllocationEEType;
```

Comment:
This can be a regular thread static in GCHelpers.cpp. I think we should avoid placing every thread static into thread.h. thread.h should only have the special thread statics that are accessed from external threads.

Reply:
Yes, I think this went a bit too far. 
It is a fancy of allocation slow path and allocation tick event. Thread does not need to report or know about this.